src/components/Dropdown.tsx

```diff
@@ -0,0 +1,106 @@
+import { useCallback, useState } from 'react';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+type Props = {
+  people: Person[],
+  searchDelay: number,
+  onSelected: (person: Person) => void,
+};
+
+function getFilteredPeople(people: Person[], query: string) {
+  if (!query) {
+    return people;
+  }
+
+  return people.filter(({ name }) => {
+    const queryLower = query.trim().toLowerCase();
+    const nameLower = name.toLowerCase();
+
+    return nameLower.includes(queryLower);
+  });
+}
+
+const FEMALE = 'f';
+const MALE = 'm';
+
+export const Dropdown: React.FC<Props> = ({
+  people,
+  searchDelay,
+  onSelected,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const personList = getFilteredPeople(people, appliedQuery);
+
+  const handlePersonClick = (person: Person) => {
+    setQuery(person.name);
+    setAppliedQuery(person.name);
+    setIsOpen(false);
+    onSelected(person);
+  };
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, searchDelay),
+    [],
+  );
+
+  const handelQueryChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
```

Comment:
You can get value with destructing and in this case, don't repeat `event.target.value` two times